openfisca_core/populations/_core_population.py

```diff
@@ -0,0 +1,455 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TypeVar
+
+import traceback
+
+import numpy
+
+from openfisca_core import holders, periods
+
+from . import types as t
+from ._errors import (
+    IncompatibleOptionsError,
+    InvalidArraySizeError,
+    InvalidOptionError,
+    PeriodValidityError,
+)
+
+#: Type variable for a covariant data type.
+_DT_co = TypeVar("_DT_co", covariant=True, bound=t.VarDType)
+
+
+class CorePopulation:
+    """Base class to build populations from.
+
+    Args:
+        entity: The :class:`.CoreEntity` of the population.
+        *__args: Variable length argument list.
+        **__kwds: Arbitrary keyword arguments.
+
+    """
+
+    #: ???
+    count: int = 0
+
+    #: The :class:`.CoreEntity` of the population.
+    entity: t.CoreEntity
+
+    #: ???
+    ids: Sequence[str] = []
+
+    #: ???
+    simulation: None | t.Simulation = None
```

Comment:
@benjello I don't understand here why entities have a tax benefit system associated, and populations have a simulation associated. Is there a design meaning to that?

Reply:
(I remember already having asking this, but I couldn't find where. It makes, however, sense to think of entity-tbs as a legal thing, and population-simulation as their _calculation interface_).